app/scripts/controllers/tokens.js

```diff
@@ -0,0 +1,231 @@
+import ObservableStore from 'obs-store'
+import { isValidAddress, toChecksumAddress } from 'ethereumjs-util'
+import { addInternalMethodPrefix } from './permissions'
+
+import {
+  ComposableController,
+  NetworkController,
+  PreferencesController,
+  AssetsDetectionController,
+  AssetsContractController,
+} from 'gaba'
+
+class Tokens {
+  constructor (opts = {}) {
+    const initState = Object.assign({
+      assetImages: {},
+      allCollectibleContracts: {},
+      allCollectibles: {},
+      allTokens: {},
+      collectibleContracts: [],
+      collectibles: [],
+      ignoredCollectibles: [],
+      ignoredTokens: [],
+      suggestedTokens: {},
+      tokens: [],
+
+    }, opts.initState)
+    this.provider = opts.provider
+    this.preferences = opts.preferences
+    this.network = opts.network
+    this.assets = opts.assets
+    this.openPopup = opts.openPopup
+    this.store = new ObservableStore(initState)
+    this.dataModel = new ComposableController([
+      new NetworkController(),
```

Comment:
This means we're running _both_ network controllers side-by-side now, including two block trackers, right? We certainly don't want to double our Infura traffic. I thought the plan was to merge the extension and gaba network controller first, before working on collectibles, to avoid this duplication.